classic_cp/kidneyexchange/kidneyexchange.jl

```diff
@@ -0,0 +1,278 @@
+using SeaPearl
+
+"""
+Instances format
+4 0.33 #numberOfPairs density
+2 4    #pair 1 can receive a kidney from pairs 2 and 4
+1      #pair 2 can receive a kidney from pair 1
+4      #pair 3 can receive a kidney from pair 4
+3      #pair 4 can receive a kidney from pair 3
+"""
+
+struct InputData
+    numberOfPairs         :: Int
+    density               :: Float16
+    compatibilities       :: Vector{Union{Vector{Int}, Nothing}}
+end
+
+include("IOmanager.jl")
+
+"""
+    Unfinished version, problem defining the objective (how to convert IntVar to Bool?)
+"""
+function solve_kidneyexchange_bis(filename::String)
+
+    InputData = getInputData(filename)
+
+    n = InputData.numberOfPairs
+    c = InputData.compatibilities
+
+    trailer = SeaPearl.Trailer()
+    model = SeaPearl.CPModel(trailer)
+
+    ### Try to reduce problem ###    
+    canReduce = true
+    current_n = n
+    while canReduce
+        flatten_c = reduce(vcat,c)
+        usefullPairs = [i for i in 1:n if i in flatten_c && c[i] != []] 
+        if length(usefullPairs) < current_n
+            c = [filter(e -> e in usefullPairs, list) for list in c]
+            current_n = length(usefullPairs)
+        else
+            canReduce = false
+        end
+    end
+
+    count = 1
+    pairsEquivalence = []
+    for i in 1:n
+        if isempty(c[i])
+            push!(pairsEquivalence, nothing)
+        else
+            push!(pairsEquivalence, count)
+            count += 1
+        end
+    end
+
+    c = filter(e -> !isempty(e), c)
+
+    for i in 1:length(c)
+        c[i] = map(e -> pairsEquivalence[e], c[i])
+        push!(c[i], i)
+    end
+
+    println("Instance reduction: original size "*string(n)*", new size "*string(current_n))
+    println()
+    n = current_n
+    
+    ### Variable declaration ###    
+    x = Vector{SeaPearl.AbstractIntVar}(undef, n) 
+    x_offset = Vector{SeaPearl.AbstractIntVar}(undef, n) 
+
+    for i = 1:n
+        x[i] = SeaPearl.IntVar(1, n, "x_"*string(i), model.trailer)
+        SeaPearl.addVariable!(model, x[i]; branchable=true)
+        x_offset[i] = SeaPearl.IntVarViewOffset(x[i], -i, x[i].id*"-"*string(i))
+    end
+
+    ### Constraints ###
+    for i = 1:n
+        for j in 1:n
+            if j ∉ c[i]
+                push!(model.constraints, SeaPearl.NotEqualConstant(x[i], j, trailer))
+            end
+        end
+    end
+    push!(model.constraints, SeaPearl.AllDifferent(x, trailer))
+
+    # additional constraint: sum(x - x_offset) = 0
+
+    ### Objective ###
+    numberOfExchanges = SeaPearl.IntVar(-n, 0, "numberOfExchanges", trailer) 
+    SeaPearl.addVariable!(model, numberOfExchanges)
+    vars = SeaPearl.AbstractIntVar[]
+    for i in 1:n
+        vars = cat(vars, int2bool(x_offset[i]); dims=1) #TODO int2bool?
+    end
+    push!(vars, numberOfExchanges)
+    objective = SeaPearl.SumToZero(vars, trailer)
+    push!(model.constraints, objective)
```

Comment:
you should not use `push!(model.constraints, your_constraint )` to add a constraint but rather `SeaPearl.addConstraint!`

Reply:
done